### ISSUE TITLE
elliptical mirror incidence angle setting 

### ISSUE THREAD

**Issue by tmazza79xfel** [lucarebuffi/ShadowOui#167]
ciao Luca,
when setting by gui the incidence angle in the "position" tab, this automatically changes the "incidence angle respect to normal" in the basic setting / surface shape tab. the opposite does not occur: so if the beam is to be propagated with a detuned incidence angle, one needs first to change the angle under "position" tab, and then change (back) the angle to the original setting in the surface shape tab.
can you please fix this?
thanks
Tommaso

**Comment by lucarebuffi** [comment]
Dear Tommaso,

What you describe is exactly how the program is supposed to work. The surface shape parameters modification do not affect the position parameters automatically. Shape and position are different concepts, indeed. I agree with you that the behaviour of the GUI should be more clear and flexible.
I think I will implement an automatic modification in both direction with a warning message, where the user can confirm the modification in both the fields.

**Comment by tmazza79xfel** [comment]
ciao Luca,

I am perfectly fine with the surface shape parameter NOT affecting the position parameter. I would be very fine with the other way around, that is the position parameter NOT affecting the surface shape. so, automatic modification in both directions is not what I favour, I would rather have that there is no modification in either of the two directions.
This is indeed the case for the source and image distances: when you change these in the position, the shape parameter remains unchanged. BUT if you change the angle in the position tab, then the surface shape parameter changes.

so there is at least inconsistency between the handling of the distance and of the angle. 
Anyhow, the implementation of a warning message would be the best way to go.

grazie
Tommaso

**Comment by lucarebuffi** [comment]
Thanks for the precious feedback. Remember that you are not the only user, indeed. I should find a solution that fits several needs simultaneously.

**Comment by srio** [comment]
Let me comment on this: I understand the inconsistence, but the behaviour is the correct: copy only in direction POSITIONS to SURFACE SHAPE and not backwards. And the expected is not to copy the distances (if this option is selected we are supposed to define different focal distances) but copy the incident angle (in principle, users do not want to change it and is a pain to enter manually). This is clear for everybody.

Perhaps the best option is change the interface and convert  "Incident angle with respect to the normal" into a pulldown/combo box "Incdent angle" with options "Copied from position" and "external". If external an additional field appears with "Incident angle with respect to the normal will appear. 

Any other solution would also be good!

**Comment by tmazza79xfel** [comment]
thanks for the clarification Manolo.
I thought the behaviour was not intended to be, that is I thought this would be a bug (notice, I have changed the title of the issue after Luca's reply).
I opened the issue because I had some bad time when I tried to detune the incidence angle in order to move the focus along the beam direction, and the focus would not move...
Now that things are clear, even the current arrangement is completely fine for me - it is enough for me to know. Yet, I think the pulldown/combo box you suggest would be the best solution.

thanks again
Tommaso
